lib/shared/src/event/error.rs

```diff
@@ -0,0 +1,48 @@
+use crate::{event::*, lookup::*};
+use std::fmt;
+pub use EventError::*;
+
+#[derive(Debug)]
+pub enum EventError {
+    PrimitiveDescent {
+        primitive_at: LookupBuf,
+        original_target: LookupBuf,
+        original_value: Option<Value>,
+    },
+    LookupError(crate::lookup::LookupError),
+    EmptyCoalesceSubSegment,
+    RemovingSelf,
+}
+
+impl fmt::Display for EventError {
```

Comment:
Could you use Snafu here to do all this?

Reply:
Indeed! I didn't really want to add in a new dependency to shared just for that though. Snafu's `ErrorCompat` enables these to use it's features anyways: https://docs.rs/snafu/0.3.1/snafu/trait.ErrorCompat.html

Would you prefer we use the lib here? Would be happy to.

Reply:
Yeah. It has been pulled in anyway in `metric/mod.rs`.

Reply:
Oh perfect!